modules/nlu/src/backend/predict-pipeline.ts

```diff
@@ -439,6 +451,16 @@ async function extractSlots(input: PredictStep, predictors: Predictors): Promise
   return { ...input, slot_predictions_per_intent: slots_per_intent }
 }
 
+async function extractElectedIntentSlots(input: PredictOutput): Promise<PredictOutput> {
+  const intentWasElectedWithoutAmbiguity = input?.intent?.name && !_.isEmpty(input.predictions) && !input.ambiguous
+  if (!intentWasElectedWithoutAmbiguity) {
+    return input
+  }
+
+  const electedIntent = _.flatMap(input.predictions, p => p.intents).find(i => i.label === input.intent.name) // might find an intent with the correct name of an incorrect context... Chosen context should maybe be in PredictOuput
```

Comment:
might find an intent with the correct name in wrong context... @EFF what do you think about putting the chosen context in PredictOutput?